addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,1024 @@
+            logger.error("An exception occurred while reading bytebuffer '{}' : {}",
+                    DatatypeConverter.printHexBinary(byteBuffer.array()), e.getMessage(), e);
+        }
+    }
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                if (socketChannel == null) {
+                    previousConnectionState = false;
+                    onConnectionLost();
+                } else {
+                    if (previousConnectionState == false && socketChannel.isConnected()) {
+                        previousConnectionState = true;
+                        onConnectionResumed();
+                    }
+
+                    if (previousConnectionState == true && !socketChannel.isConnected()
+                            && !socketChannel.isConnectionPending()) {
+                        previousConnectionState = false;
+                        onConnectionLost();
+                    }
+
+                    if (!socketChannel.isConnectionPending() && !socketChannel.isConnected()) {
+                        previousConnectionState = false;
+                        logger.debug("Disconnecting '{}' because of a network error", getThing().getUID().toString());
+                        disconnect(socketChannel);
+                        Thread.sleep(1000 * ((BigDecimal) getConfig().get(RECONNECT_INTERVAL)).intValue());
+                        establishConnection();
+                    }
+
+                    long stamp = System.currentTimeMillis();
+                    if (!InetAddress.getByName(((String) getConfig().get(IP_ADDRESS)))
+                            .isReachable(((BigDecimal) getConfig().get(PING_TIME_OUT)).intValue())) {
+                        logger.debug(
+                                "Ping timed out after '{}' milliseconds. Disconnecting '{}' because of a ping timeout",
+                                System.currentTimeMillis() - stamp, getThing().getUID().toString());
+                        disconnect(socketChannel);
+                    }
+
+                    onConnectable(socketChannel);
+                    ByteBuffer buffer = onReadable(socketChannel,
+                            ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+                    if (buffer != null && buffer.remaining() > 0) {
+                        onRead(buffer, socketChannel);
+                    }
+                }
+
+                onAcceptable(listenerChannel);
+
+            } catch (Exception e) {
+                logger.trace("An exception occurred while polling the transceiver : '{}'", e.getMessage(), e);
+            }
+        }
+    };
+
+    public boolean registerTransceiverStatusListener(TransceiverStatusListener transceiverStatusListener) {
+        if (transceiverStatusListener == null) {
+            throw new NullPointerException("It's not allowed to pass a null BlasterStatusListener.");
```

Comment:
It's a best practice not to throw the NPE yourselves, there is however an alternative which is also shorter: https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull(T,%20java.lang.String)